IdentityCore/src/cache/token/MSIDLegacyTokenCacheItem.m

```diff
@@ -123,12 +126,19 @@ - (instancetype)initWithCoder:(NSCoder *)coder
     self.cachedAt = [coder decodeObjectOfClass:[NSDate class] forKey:@"cachedAt"];
     self.familyId = [coder decodeObjectOfClass:[NSString class] forKey:@"familyId"];
 
-    self.additionalInfo = [coder decodeObjectOfClass:[NSDictionary class] forKey:@"additionalServer"];
-    if(self.additionalInfo[MSID_EXTENDED_EXPIRES_ON_CACHE_KEY])
+    NSMutableDictionary *additionalServer = [[coder decodeObjectOfClass:[NSDictionary class] forKey:@"additionalServer"] mutableCopy];
+    if(additionalServer[MSID_EXTENDED_EXPIRES_ON_CACHE_KEY])
     {
-        self.extendedExpiresOn = self.additionalInfo[MSID_EXTENDED_EXPIRES_ON_CACHE_KEY];
-        NSMutableDictionary *additionalServer = [[NSMutableDictionary alloc] initWithDictionary:self.additionalInfo];
+        self.extendedExpiresOn = additionalServer[MSID_EXTENDED_EXPIRES_ON_CACHE_KEY];
         [additionalServer removeObjectForKey:MSID_EXTENDED_EXPIRES_ON_CACHE_KEY];
+    }
+    if(additionalServer[MSID_SPE_INFO_CACHE_KEY])
+    {
+        self.speInfo = additionalServer[MSID_SPE_INFO_CACHE_KEY];
+        [additionalServer removeObjectForKey:MSID_SPE_INFO_CACHE_KEY];
```

Comment:
I think ADAL is just looking into additional info. Setting spe info separately would require updates in ADAL.